### ISSUE TITLE
[oldenglish picker] Bugs in oldEnglishToIPA

### ISSUE THREAD

**Issue by dscorbett** [r12a/pickers#52]
⟨|⟩ gets converted to ⟨ð⟩.

⟨[]{⟩ gets converted to ⟨&#ks5b;&#ks5d;&#ks7b;⟩.

The rare letter ⟨k⟩ isn’t treated like ⟨c⟩ in ⟨nk⟩, which becomes ⟨nk⟩ instead of ⟨ŋk⟩.

The voiceless geminate ⟨þþ⟩ or ⟨ðð⟩ becomes ⟨ðð⟩.

The rare digraph ⟨qu⟩ isn’t supported.

⟨nġ⟩ should become ⟨nʤ⟩.

The diphthongs ⟨ēa⟩, ⟨ēo⟩, and ⟨īe⟩ are transcribed with U+003A COLON instead of U+02D0 MODIFIER LETTER TRIANGULAR COLON.

**Comment by r12a** [comment]
Thanks for the report!  I've addressed some of the easy things.  I need some time to check the phonetic rules, so i'll come back to those later.

> The diphthongs ⟨ēa⟩, ⟨ēo⟩, and ⟨īe⟩ are transcribed with U+003A COLON instead of U+02D0 MODIFIER LETTER TRIANGULAR COLON.

fixed

> ⟨[]{⟩ gets converted to ⟨&#ks5b;&#ks5d;&#ks7b;⟩.

hmm. This is down to some code that escapes such characters 'for security', then transforms the escapes.  I need to check that out in more detail.

> The voiceless geminate ⟨þþ⟩ or ⟨ðð⟩ becomes ⟨ðð⟩.

By 'becomes' do you mean 'should become'?  Because i get 'θθ' for both of those.

**Comment by moyogo** [comment]
As a side note, the IPA deprecated the affricate ligature symbols like ʤ in 1989, dʒ or d͡ʒ should be used instead.

**Comment by dscorbett** [comment]
> > The voiceless geminate ⟨þþ⟩ or ⟨ðð⟩ becomes ⟨ðð⟩.
> 
> By 'becomes' do you mean 'should become'? Because i get 'θθ' for both of those.

That is true in isolation, but not within a word. For example, ⟨oþþe⟩ is converted to ⟨oððe⟩ but should be converted to ⟨oθθe⟩.

**Comment by r12a** [comment]
@moyogo see https://github.com/r12a/pickers/issues/1 wrt your side note.
